tests/snowpark/test_packages_available_in_snowflake.py

```diff
@@ -0,0 +1,222 @@
+import json
+from unittest import mock
+from unittest.mock import MagicMock, patch
+
+import pytest
+from snowflake.cli.plugins.snowpark.models import Requirement
+from snowflake.cli.plugins.snowpark.package.packages_in_snowflake import (
+    AvailablePackage,
+    PackagesAvailableInSnowflake,
+    PackagesAvailableInSnowflakeManager,
+)
+from snowflake.connector import Error as ConnectorError
+
+from tests.snowpark.mocks import mock_available_packages_sql_result  # noqa: F401
+from tests.test_data import test_data
+from tests.testing_utils.fixtures import TEST_DIR
+
+PACKAGES_AVAILABLE_IN_SNOWFLAKE = PackagesAvailableInSnowflake(
+    packages={
+        "shrubbery": AvailablePackage(
+            snowflake_name="shrubbery", versions={"1.2.1", "1.2.2"}
+        ),
+        "dummy_pkg": AvailablePackage(
+            snowflake_name="dummy-pkg", versions={"0.1.1", "1.0.0", "1.1.0"}
+        ),
+        "jpeg": AvailablePackage(snowflake_name="jpeg", versions={"9e", "9d", "9b"}),
+    }
+)
+
+
+def test_latest_version():
+    assert (
+        PACKAGES_AVAILABLE_IN_SNOWFLAKE.package_latest_version(
+            Requirement.parse("shrubbery")
+        )
+        == "1.2.2"
+    )
+    assert (
+        PACKAGES_AVAILABLE_IN_SNOWFLAKE.package_latest_version(
+            Requirement.parse("dummy_pkg")
+        )
+        == "1.1.0"
+    )
+    assert (
+        PACKAGES_AVAILABLE_IN_SNOWFLAKE.package_latest_version(
+            Requirement.parse("dummy-pkg")
+        )
+        == "1.1.0"
+    )
+    assert (
+        PACKAGES_AVAILABLE_IN_SNOWFLAKE.package_latest_version(
+            Requirement.parse("jpeg")
+        )
+        is None
+    )
+    assert (
+        PACKAGES_AVAILABLE_IN_SNOWFLAKE.package_latest_version(
+            Requirement.parse("weird-pkg")
+        )
+        is None
+    )
+
+
+def test_versions():
+    assert PACKAGES_AVAILABLE_IN_SNOWFLAKE.package_versions(
+        Requirement.parse("shrubbery")
+    ) == [
+        "1.2.2",
+        "1.2.1",
+    ]
+    assert PACKAGES_AVAILABLE_IN_SNOWFLAKE.package_versions(
+        Requirement.parse("dummy_pkg")
+    ) == [
+        "1.1.0",
+        "1.0.0",
+        "0.1.1",
+    ]
+    assert PACKAGES_AVAILABLE_IN_SNOWFLAKE.package_versions(
+        Requirement.parse("dummy-pkg")
+    ) == [
+        "1.1.0",
+        "1.0.0",
+        "0.1.1",
+    ]
+    assert PACKAGES_AVAILABLE_IN_SNOWFLAKE.package_versions(
+        Requirement.parse("jpeg")
+    ) == ["9e", "9d", "9b"]
+    assert (
+        PACKAGES_AVAILABLE_IN_SNOWFLAKE.package_versions(Requirement.parse("weird-pkg"))
+        == []
+    )
```

Comment:
Why not pytest.mark.parametrize?

Reply:
This is the old test (from `test_anaconda.py`).